lib/fizzy/parser_expr.cpp

```diff
@@ -500,30 +496,27 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, FuncIdx f
 
             update_result_stack(frame, operand_stack);  // else is the end of if.
 
-            const auto if_imm_offset = frame.immediates_offset;
+            const auto if_imm_offset = frame.code_offset + 1;
             const auto frame_type = frame.type;
             auto frame_br_immediate_offsets = std::move(frame.br_immediate_offsets);
 
             control_stack.pop();
             control_stack.emplace(Instr::else_, frame_type, static_cast<int>(operand_stack.size()),
-                code.instructions.size(), code.immediates.size());
+                code.instructions.size());
             // br immediates from `then` branch will need to be filled at the end of `else`
             control_stack.top().br_immediate_offsets = std::move(frame_br_immediate_offsets);
 
             // Placeholders for immediate values, filled at the matching end instructions.
```

Comment:
Move this comment one line down, to before pushing the immediate
```suggestion
            // Placeholder for immediate value, filled at the matching end instructions.
```